test/uk/gov/hmrc/gform/sharedmodel/formtemplate/generators/DestinationsGen.scala

```diff
@@ -29,12 +31,24 @@ trait DestinationsGen {
       .oneOrMoreGen(DestinationGen.destinationGen)
       .map(Destinations.DestinationList(_, ackSection, decSection))
 
-  def printSectionGen: Gen[Destinations.PrintSection] =
-    PrimitiveGen
-      .oneOrMoreGen(Gen.const(PrintSection(toSmartString("TestTitle"), toSmartString("TestSummaryPdf"))))
-      .map(_.head)
+  def destinationPrintGen: Gen[DestinationPrint] =
+    for {
+      pdf <- Gen.option(
+              Pdf(
+                toSmartString("TestHeader"),
+                toSmartString("TestFooter"),
+                List(FormComponentId("TestId1"), FormComponentId("TestId2"))))
+
+      destinationPrint <- PrimitiveGen
+                           .oneOrMoreGen(
```

Comment:
`oneOrMoreGen(...).map(_.head)` seems like waste of CPU resources.

Reply:
We need random `Gen`s for `PrintSection.Pdf` and `PrintSection.Page`.

Once we have them than `def destinationPrintGen: Gen[DestinationPrint] = ...` will implemented in terms of them.